### PR TITLE
fix Location#== with nil

### PR DESCRIPTION
**Pull request by joevandyk**
`location == nil` will blow up. This fixes that.

(We had code like `location != nil` for some reason in our codebase)

### CODE REVIEW

**Comment by joevandyk** [comment]
@tamcgoey @garethson any chance this could get reviewed?

File: lib/active_shipping/location.rb [hunk 1]
```diff
@@ -160,6 +160,7 @@ def address2_and_3
     end
 
     def ==(other)
+      return false if !other.respond_to?(:to_hash)
```
Comment (javierhonduco) [inline lib/active_shipping/location.rb]: I'd go for something like
```patch
     def ==(other)
-      to_hash == other.to_hash
+      to_hash == other&.to_hash
     end
```

File: test/unit/location_test.rb [hunk 2]
```diff
@@ -231,4 +231,8 @@ def province_code
     assert_equal @location, another_location
     refute_equal @location, Location.new({})
   end
+
+  test "#== works with nil" do
+    refute_equal @location, nil
```
Comment (javierhonduco) [inline test/unit/location_test.rb]: `refute_nil` could be used here

File: lib/active_shipping/location.rb (hunk 1 above)
Comment (joevandyk) [reply lib/active_shipping/location.rb]: That’s a newer ruby feature.. is that ok to do?

Comment (joevandyk) [reply lib/active_shipping/location.rb]: I think `&.` was introduced in Ruby 2.3 and Travis is still testing 2.2.

Comment (javierhonduco) [reply lib/active_shipping/location.rb]: Not sure about that! Ruby 2.3 was introduced 2 years ago, but probably Thomas has an answer :)

File: test/unit/location_test.rb (hunk 2 above)
Comment (javierhonduco) [reply test/unit/location_test.rb]: My bad, using `refute_nil` is not ok, as we don't want to check that the value is not `nil` (which I think should not happen), but that the comparison with `nil` returns `false` 😅

File: lib/active_shipping/location.rb (hunk 1 above)
Comment (tamcgoey) [reply lib/active_shipping/location.rb]: Yea, we wouldn't be able to use the safe navigator operator here, since `ActiveShipping` is still supporting 2.2.

**Comment by tamcgoey** [comment]
👋  @joevandyk, sorry for the delay getting to this PR. As of [v2.0 we're not actively maintaining](https://github.com/Shopify/active_shipping#versions) this gem anymore, but we're always looking for maintainers.

**Pull request closed without merging**